core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -252,26 +247,28 @@ export class TopicEditorStateService {
       canonicalStorySummaries,
       newBackendTopicRightsObject
     ]) => {
-      this._skillCreationIsAllowed = (
-        newBackendTopicObject.skillCreationIsAllowed);
-      this._skillQuestionCountDict = (
-        newBackendTopicObject.skillQuestionCountDict);
-      this._updateGroupedSkillSummaries(
-        newBackendTopicObject.groupedSkillSummaries);
       this._updateTopic(
         newBackendTopicObject.topicDict,
         newBackendTopicObject.skillIdToDescriptionDict
       );
-      this._updateGroupedSkillSummaries(
-        newBackendTopicObject.groupedSkillSummaries);
-      this._updateSkillIdToRubricsObject(
-        newBackendTopicObject.skillIdToRubricsDict);
-      this._updateClassroomUrlFragment(
-        newBackendTopicObject.classroomUrlFragment);
-      this._updateTopicRights(newBackendTopicRightsObject);
-      this._setCanonicalStorySummaries(canonicalStorySummaries);
-      this._topicIsLoading = false;
-      this.loaderService.hideLoadingScreen();
+      if (this._topic) {
```

Comment:
At this point, the promise has executed and the topic data is present.

So why would it still be the case here that this._topic is not defined?

Reply:
Done, this line has been removed